src/libraries/System.Net.Sockets/tests/FunctionalTests/SendReceive/SendReceive.cs

```diff
@@ -1149,6 +1182,39 @@ public sealed class SendReceive_Eap : SendReceive<SocketHelperEap>
     public sealed class SendReceive_SpanSync : SendReceive<SocketHelperSpanSync>
     {
         public SendReceive_SpanSync(ITestOutputHelper output) : base(output) { }
+
+        [Fact]
+        public async Task Send_0ByteSend_Span_Success()
+        {
+            using (Socket listener = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp))
+            using (Socket client = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp))
+            {
+                listener.Bind(new IPEndPoint(IPAddress.Loopback, 0));
+                listener.Listen(1);
+
+                Task<Socket> acceptTask = AcceptAsync(listener);
+                await Task.WhenAll(
+                    acceptTask,
+                    ConnectAsync(client, new IPEndPoint(IPAddress.Loopback, ((IPEndPoint)listener.LocalEndPoint).Port)));
+
+                using (Socket server = await acceptTask)
+                {
+                    for (int i = 0; i < 3; i++)
```

Comment:
Just wondering: what is the behavior that makes it worth running the same test 3 times on the same socket?

Reply:
Nothing in particular; I was mimicking other tests that do similar stuff. I can remove if you think it's weird.